src/core/src/node.cpp

```diff
@@ -282,8 +282,13 @@ std::string ov::Node::description() const {
 }
 
 const std::string& ov::Node::get_friendly_name() const {
+    const auto& name = get_name();
+    AtomicGuard lock(m_name_changing);
     if (m_friendly_name.empty()) {
-        return get_name();
+        if (m_auto_generated_friendly_name.empty()) {
+            m_auto_generated_friendly_name = name + "_" + "autogenerated";
```

Comment:
```suggestion
            m_auto_generated_friendly_name = get_bane() + "_" + "autogenerated";
```
And remove `name` reference from global scope of this function.